File: bestia-models/src/test/java/net/bestia/model/map/TilesetTest.java
package net.bestia.model.map;

import org.junit.Assert;
import org.junit.Test;

import net.bestia.model.geometry.Size;
import net.bestia.model.map.TileProperties;
import net.bestia.model.map.Tileset;


public class TilesetTest {

	@Test(expected=IllegalArgumentException.class)
	public void setProperties_invalidGid_throws() {
		Tileset set = getTileset();
		set.setProperties(1234, new TileProperties(true, 100, false));
	}

	@Test
	public void setProperties_validGid_ok() {
		Tileset set = getTileset();
		set.setProperties(145, new TileProperties(true, 100, false));
		set.setProperties(100, new TileProperties(true, 100, false));
		set.setProperties(200, new TileProperties(true, 100, false));
	}

	@Test(expected=NullPointerException.class)
	public void setProperties_nullProperties_throws() {
		Tileset set = getTileset();
		set.setProperties(145, null);
	}
	
	@Test
	public void contains_validGid_true() {
		Tileset set = getTileset();
		Assert.assertTrue(set.contains(100));
		Assert.assertTrue(set.contains(156));
		Assert.assertTrue(set.contains(200));
	}
	
	@Test
	public void contains_invalidGid_false() {
		Tileset set = getTileset();
		Assert.assertFalse(set.contains(1));
		Assert.assertFalse(set.contains(99));
		Assert.assertFalse(set.contains(201));
	}

	public Tileset getTileset() {
		return new Tileset("test", new Size(320, 320), 100);
	}
}

File: bestia-models/src/test/kotlin/net/bestia/model/map/TilesetTest.kt
package net.bestia.model.map

import net.bestia.model.geometry.Size
import org.junit.Assert
import org.junit.Test

class TilesetTest {

  val tileset: Tileset
    get() = Tileset("test", Size(10, 10), 100)

  @Test(expected = IllegalArgumentException::class)
  fun setProperties_invalidGid_throws() {
    val set = tileset
    set.setProperties(1234, TileProperties(true, 100, false))
  }

  @Test
  fun setProperties_validGid_ok() {
    val set = tileset
    set.setProperties(145, TileProperties(true, 100, false))
    set.setProperties(100, TileProperties(true, 100, false))
    set.setProperties(200, TileProperties(true, 100, false))
  }

  @Test
  fun contains_validGid_true() {
    val set = tileset
    Assert.assertTrue(set.contains(100))
    Assert.assertTrue(set.contains(156))
    Assert.assertTrue(set.contains(200))
  }

  @Test
  fun contains_invalidGid_false() {
    val set = tileset
    Assert.assertFalse(set.contains(1))
    Assert.assertFalse(set.contains(99))
    Assert.assertFalse(set.contains(201))
  }
}
